PhysicsTools/PatExamples/plugins/MuonVIDCITKAnalyzer.cc

```diff
@@ -0,0 +1,206 @@
+
+#include "TH1F.h"
+#include "TH2F.h"
+
+#include <sstream>
+
+using namespace std;
+using namespace edm;
+//
+// class decleration
+//
+class MuonVIDCITKAnalyzer : public edm::stream::EDAnalyzer<>
+{
+public:
+  MuonVIDCITKAnalyzer(const edm::ParameterSet&);
+  void analyze(const edm::Event&, const edm::EventSetup&) override;
+
+private:
+  edm::EDGetTokenT<edm::View<reco::Candidate> > muonToken_;
+  edm::EDGetTokenT<reco::VertexCollection> vertexToken_;
+
+  // VID
+  typedef edm::ValueMap<bool> VIDMap;
+  edm::EDGetTokenT<VIDMap> muonLooseVIDToken_;
+  edm::EDGetTokenT<VIDMap> muonMediumVIDToken_;
+  edm::EDGetTokenT<VIDMap> muonTightVIDToken_;
+  edm::EDGetTokenT<VIDMap> muonSoftVIDToken_;
+  edm::EDGetTokenT<VIDMap> muonHighPtVIDToken_;
+
+  // CITK
+  typedef edm::ValueMap<float> CITKMap;
+  edm::EDGetTokenT<CITKMap> muonChIsoCITKToken_;
+  edm::EDGetTokenT<CITKMap> muonNhIsoCITKToken_;
+  edm::EDGetTokenT<CITKMap> muonPhIsoCITKToken_;
+  edm::EDGetTokenT<CITKMap> muonPuIsoCITKToken_;
+
+  // IsoDeposit
+  typedef edm::ValueMap<double> IsoMap;
+  edm::EDGetTokenT<IsoMap> muonChIsoIsoDepToken_;
+  edm::EDGetTokenT<IsoMap> muonNhIsoIsoDepToken_;
+  edm::EDGetTokenT<IsoMap> muonPhIsoIsoDepToken_;
+  edm::EDGetTokenT<IsoMap> muonPuIsoIsoDepToken_;
+
+  // Histograms
+  TH2F* h2Ch_, * h2Nh_, * h2Ph_, * h2Pu_;
+  TH1F* hIsoDiffCh_, * hIsoDiffNh_, * hIsoDiffPh_, * hIsoDiffPu_;
+};
+
+MuonVIDCITKAnalyzer::MuonVIDCITKAnalyzer(const edm::ParameterSet& iConfig)
+{
+  muonToken_ = consumes<edm::View<reco::Candidate> >(edm::InputTag("muons"));
+  vertexToken_ = consumes<reco::VertexCollection>(edm::InputTag("offlinePrimaryVertices"));
+
+  string vidPrefix = "muoMuonIDs:cutBasedMuonId-MuonPOG-V0-";
+  muonLooseVIDToken_  = consumes<VIDMap>(edm::InputTag(vidPrefix+"loose" ));
+  muonMediumVIDToken_ = consumes<VIDMap>(edm::InputTag(vidPrefix+"medium"));
+  muonTightVIDToken_  = consumes<VIDMap>(edm::InputTag(vidPrefix+"tight" ));
+  muonSoftVIDToken_   = consumes<VIDMap>(edm::InputTag(vidPrefix+"soft"  ));
+  muonHighPtVIDToken_ = consumes<VIDMap>(edm::InputTag(vidPrefix+"highpt"));
+
+  muonChIsoCITKToken_ = consumes<CITKMap>(edm::InputTag("muonPFNoPileUpIsolation:h+-DR040-ThresholdVeto000-ConeVeto000"));
+  muonNhIsoCITKToken_ = consumes<CITKMap>(edm::InputTag("muonPFNoPileUpIsolation:h0-DR040-ThresholdVeto050-ConeVeto000"));
+  muonPhIsoCITKToken_ = consumes<CITKMap>(edm::InputTag("muonPFNoPileUpIsolation:gamma-DR040-ThresholdVeto050-ConeVeto000"));
+  muonPuIsoCITKToken_ = consumes<CITKMap>(edm::InputTag("muonPFPileUpIsolation:h+-DR040-ThresholdVeto050-ConeVeto000"));
+
+  muonChIsoIsoDepToken_ = consumes<IsoMap>(edm::InputTag("muPFIsoValueCharged04PAT"));
+  muonNhIsoIsoDepToken_ = consumes<IsoMap>(edm::InputTag("muPFIsoValueNeutral04PAT"));
+  muonPhIsoIsoDepToken_ = consumes<IsoMap>(edm::InputTag("muPFIsoValueGamma04PAT"));
+  muonPuIsoIsoDepToken_ = consumes<IsoMap>(edm::InputTag("muPFIsoValuePU04PAT"));
+
+  edm::Service<TFileService> fs;
```

Comment:
Hi Junghwan,

This analyzer will never be used as part of the reconstruction chain, only
in test cases.
There is no need to bloat the plugins library with it, would prefer that it
is moved to test.

It would be useful to move only those analyzers that don't make sense as
part of the reconstruction chain into test and then build individual plugin
libraries there.

Best,
-Lindsey

On Wed, Jun 10, 2015 at 11:25 AM, Junghwan John Goh <
notifications@github.com> wrote:

> In PhysicsTools/PatExamples/plugins/MuonVIDCITKAnalyzer.cc
> <https://github.com/cms-sw/cmssw/pull/9459#discussion_r32101032>:
>
> > +  muonMediumVIDToken_ = consumes<VIDMap>(edm::InputTag(vidPrefix+"medium"));
> > +  muonTightVIDToken_  = consumes<VIDMap>(edm::InputTag(vidPrefix+"tight" ));
> > +  muonSoftVIDToken_   = consumes<VIDMap>(edm::InputTag(vidPrefix+"soft"  ));
> > +  muonHighPtVIDToken_ = consumes<VIDMap>(edm::InputTag(vidPrefix+"highpt"));
> > +
> > +  muonChIsoCITKToken_ = consumes<CITKMap>(edm::InputTag("muonPFNoPileUpIsolation:h+-DR040-ThresholdVeto000-ConeVeto000"));
> > +  muonNhIsoCITKToken_ = consumes<CITKMap>(edm::InputTag("muonPFNoPileUpIsolation:h0-DR040-ThresholdVeto050-ConeVeto000"));
> > +  muonPhIsoCITKToken_ = consumes<CITKMap>(edm::InputTag("muonPFNoPileUpIsolation:gamma-DR040-ThresholdVeto050-ConeVeto000"));
> > +  muonPuIsoCITKToken_ = consumes<CITKMap>(edm::InputTag("muonPFPileUpIsolation:h+-DR040-ThresholdVeto050-ConeVeto000"));
> > +
> > +  muonChIsoIsoDepToken_ = consumes<IsoMap>(edm::InputTag("muPFIsoValueCharged04PAT"));
> > +  muonNhIsoIsoDepToken_ = consumes<IsoMap>(edm::InputTag("muPFIsoValueNeutral04PAT"));
> > +  muonPhIsoIsoDepToken_ = consumes<IsoMap>(edm::InputTag("muPFIsoValueGamma04PAT"));
> > +  muonPuIsoIsoDepToken_ = consumes<IsoMap>(edm::InputTag("muPFIsoValuePU04PAT"));
> > +
> > +  edm::Service<TFileService> fs;
>
> There are other example analyzer module in the same directory. I don't
> think all of them to be moved to test directory. I think the simplest
> solution is to change this module back to usual EDAnalyzer.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/9459/files#r32101032>.
>